interview-questions.md

```diff
@@ -7,90 +7,97 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 
 6. What is yarn? What file(s) are updated when you run `$ yarn`?
 
-  Your answer:
+  Your answer: Yarn is a Javascript packages manager. When $ yarn is ran it updates/generates your node-modules.
 
-  Researched answer:
+  Researched answer: Yarn is a new JavaScript package manager that aims for speed, and reliability. It allows you to use and share code with other developers. It is shared through a package. More specifically package.json that has all the code being shared. All the packages installed are often referred to as "modules". It was also developed by the almighty Facebook. Running '$ yarn' will default to 'yarn install" this will install the dependencies defined in a package.json file. 
 
 
 
 7. What is localhost:3000? How do we use localhost:3000 in the development process?
 
-  Your answer:
+  Your answer: Localhost:3000 is the local port used to display and run our React app on our local host server (our computer only). We use it so we can view our React app all compiled in our web browser. It will also display any errors if it runs into any while being compiled. 
 
-  Researched answer:
+  Researched answer: Localhost:3000 is a network address that points to your computer.Is is the port that React apps automatically connect to when you run "yarn start" in the terminal. It will be open in a new window/tab of your internet browser. It gives you a real-time view of your app as it is so far. It also will automatically update with any changes that you save in your react app files. So long as it is compiled correctly, passes to the Virtual DOM, and updated to the real DOM you will see the changes. This allows us to see our application without paying for a hosting service or any other backend work while still early on in the development/production process.  
 
 
 
 8. What is an iterable in JavaScript? What data types are iterables? How can this information useful to you as a developer?
 
-  Your answer:
+  Your answer: An iterable object is an object that defines a sequence and returns something. A data type that is considered a iterable are arrays. Most of these utilize built-in methods to help streamline your code. 
+
+  Researched answer: An iterable is any object that returns a function that produces and iterator for its symbol. You can loop over all values in an iterable object by using a 'for loop'. Data types that are considered to be iterables are: arrays, strings, maps, and sets. Iterables are helpful because it can be difficult to manage the iteration on data for various types of data structures.  
 
-  Researched answer:
 
 
 
 9. STRETCH: What is hoisting in JavaScript?
 
-  Your answer:
+  Your answer: I'm not sure what hoisting is in Javascript. However, I do know that when you hoist up a sail you are lifting it up to the top of the mast. So, perhaps it involves putting something to top of say perhaps a function? I will be sure to research a more complete answer and get back to you.
```

Comment:
Ooh, you're so close! Good thinking about the idea of lifting something to the top. 

I know you've learned what hoisting is in your researched answer- something I want to mention about Hoisting is that its generally bad practice to do. Just keep that in mind, we want our variables above our functions before they are used.